include/conduit/conduit.h

```diff
@@ -488,7 +498,16 @@ struct ChannelInterface<R(T...)>
     using signature_return_type = R;
 
     // Make sure we're a POD.
-    ChannelInterface() = default;
+    #ifdef CONDUIT_SOURCE_STRING_INTERNING
+    ChannelInterface(uint64_t id, Channel<R(T...)> *c) : source_id(id), channel(c)
```

Comment:
A non-defaulted user defined constructor no longer makes this a POD. Should either take out the constructors or remove the comment.
Should also default the missing copy/assignment operators probably.